include/integratorxx/quadratures/gausslegendre.hpp

```diff
@@ -96,17 +96,29 @@ struct quadrature_traits<
         point_type p_n, dp_n;
 
         // Solve the root to eps absolute precision
-        while(std::abs(z-z_old) > eps) {
+        bool converged = false;
+        const int maxit = 100;
+        for(int it = 0; it < maxit; ++it) {
           // Evaluate the Legendre polynomial at z and its derivative
-          auto pn_eval = eval_Pn(z,npts);
-          p_n = pn_eval.first;
-          dp_n = pn_eval.second;
+          std::tie(p_n, dp_n) = eval_Pn(z,npts);
 
           // Newton update for root
           z_old = z;
           z  = z_old - p_n / dp_n;
```

Comment:
```suggestion
          z  -= p_n / dp_n;
```

Reply:
Does `z` always contain what was previously in `z_old`? Working from mobile, so it's hard to tell.

Reply:
Yes, `z_old` is set to `z_old=z` on the preceding line.